exporters/holstebro_helpers.py

```diff
@@ -0,0 +1,811 @@
+                               'locale': 'da',
+                               'tmp_password': 'true',
+                               'password': 'abcd1234'
+                               }
+
+                        rows.append(row)
+                    else:
+                        logger.debug(
+                            f"User not exported to essenslms. Uuid and userkey: {uuid}, {handle}")
+
+    my_options = {"extrasaction": "ignore",
+                  "delimiter": ",",
+                  "quoting": csv.QUOTE_MINIMAL
+                  }
+
+    logger.info(f"Writing {len(rows)} rows to {filename}")
+    mh._write_csv(fieldnames, rows, filename, **my_options)
+
+
+def export_to_planorama(mh, all_nodes, filename_org, filename_persons, org_root_uuid):
+    """ Traverses a tree of OUs, for each OU finds the manager of the OU.
+    : param nodes: The nodes of the OU tree
+    """
+    fieldnames_persons = ['integrationsid',
+                          'Username',
+                          'Name',
+                          'Title',
+                          'Address',
+                          'Email',
+                          'Mobile',
+                          'Telephone',
+                          'Responsible',
+                          'Company']
+    fieldnames_org = ['Root', 'Number', 'Name', 'Manager']
+
+    rows_org = []
+    rows_persons = []
+
+    roots = SETTINGS['exports.holstebro.roots']
+    org_root_name = SETTINGS['municipality.name']
+    # insert root
+    row_org = {'Root': '', 'Number': org_root_uuid, 'Name': org_root_name}
+    rows_org.append(row_org)
+    logger.info(f"Exporting root unit: {org_root_name} to Planorama")
+
+    for root_uuid, nodes in all_nodes.items():
+
+        # is this a partail export
+        export_all = True if len(roots[root_uuid]) == 0 else False
+
+        if not export_all:
+            # partial export
+            # insert this root_uuid in rows_org with holstebro_uuid as parent
+            t_node = Node(root_uuid)
+            ou = mh.read_ou(root_uuid)
+
+            # find this unit's manager
+            manager = find_org_manager(mh, t_node)
+
+            row_org = {'Root': org_root_uuid,
+                       'Number': ou['uuid'],
+                       'Name': ou['name'],
+                       'Manager': manager['uuid']}
+
+            rows_org.append(row_org)
+            logger.info(f"Exporting {ou['name']} to Planorama")
+
+        for node in PreOrderIter(nodes['root']):
+            # if this is only partial export of org and this org unit is not on the exports list
+            # or is a child of an org unit in the exports list, continue
+            export_node = False
+            node_parent = node
+            while(node_parent != None and export_node is False):
+                export_node = True if node_parent.name in roots[root_uuid] else False
+                node_parent = node_parent.parent  # look backwards in hierarchy
+
```

Comment:
4 duplikater af det her. Der er en tommelfinger regl der siger at det er okay at have to kopier, men ved 3 skal man refaktorisere og splitte det ud: https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)